todo.sh

```diff
@@ -1113,6 +1118,50 @@ case $action in
     IFS=$SAVEIFS
     ;;
 
+"addr")
+    if [[ -z "$2" && $TODOTXT_FORCE = 0 ]]; then
+        echo -n "Add: "
+        read input
+    else
+        [ -z "$2" ] && die "usage: $TODO_SH addr \"TODO ITEM\""
+        shift
+        input=$*
+    fi
+
+	# get the last line added and extract context
+	lastadded=$(sed -n '$ =' "$TODO_FILE")
+	# FIXME: Find a better way to add tasks (_addto function?)
+	"$TODO_FULL_SH" command add "$1"
+	newest=$(sed -n '$ =' "$TODO_FILE")
+
+	REGEX='[@|+][A-Za-z0-9]*'
+
+	echo "Appending:" $(sed $lastadded"q;d" "$TODO_FILE" | grep -o "$REGEX")
+
+	# add contexts and projects
+	sed $lastadded"q;d" "$TODO_FILE" | grep -o "$REGEX" | while read -r app
+	do
+		echo $app
+	# FIXME: Find a better way to append content
```

Comment:
Again, don't do a recursive call. Either duplicate the `sed` call found in the `append` implementation, or (better) extract a new function and use that for both.